mw_basic_auth.go

```diff
@@ -137,24 +139,23 @@ func (k *BasicAuthKeyIsValid) compareHashAndPassword(hash string, password strin
 		}
 
 		if cacheEnabled {
+
 			logEntry.Debug("cache enabled: save")
-			k.cache.Set(hashedPassword, password, cacheDuration)
+			k.cache.Set(hashedPassword, string(hasher.Sum([]byte(password))), cacheDuration)
```

Comment:
I think `hasher.Sum([]byte(password))` will give you the same `password`,
need to be
```
hasher := New32()
hasher.Write(password)
hasher.Sum32()
k.cache.Set(hashedPassword, string(hasher.Sum(nil)), cacheDuration)
```

Reply:
Yeah - I think i messed that up... good spot